---
layout: post
title: "MultiProcess Pickle"
date: 2021-10-30
categories: TIL MultiProcess
---

- **Error Message**
Can't pickle local object ...

로컬 객체로 멀티프로세싱을 실행하면 이러한 오류가 발생한다

- **map_async**

```python
from multiprocessing import Pool

def callback_func(result):
    print("callback_func got result :",result)

with Pool(PROCESS_COUNT) as p:
		p.map_async(fuc, [1,2,3,4,5], callback=callback_func).get() 
```
첫번때 인자값은 수행할 function  
두번째 인자값으로 input이 들어간다  
callback은 콜백함수를 받는다  

iterable한 인자값이 들어가므로 메모리 소모가 적다
